calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java

```diff
@@ -5925,85 +5933,47 @@ else if (calendarObj.getGroupsAllowGetEvent().size() > 0)
 		boolean dateDsc = sstate.getAttribute(STATE_DATE_SORT_DSC) != null;
 		context.put("currentDateSortAsc", Boolean.valueOf(!dateDsc));
 		
-		if (!dateDsc)
-		{
-			for (yearInt = CalendarFilter.LIST_VIEW_STARTING_YEAR;
-				yearInt <= CalendarFilter.LIST_VIEW_ENDING_YEAR;
-				yearInt++)
-			{
-				Vector arrayOfMonths = new Vector(20);
-				for(monthInt = 1; monthInt <13; monthInt++)
-				{
-					CalendarUtil AcalObj = new CalendarUtil();
-					
-					monthObj2 = new MyMonth();
-					AcalObj.setDay(yearInt, monthInt, dayInt);
-					
-					dateObj1.setTodayDate(AcalObj.getMonthInteger(),AcalObj.getDayOfMonth(),AcalObj.getYear());
-					
-					// Get the events for the particular month from the
-					// master list of events.
-					calendarEventVectorObj =
-						new CalendarEventVector(
-							state.getCalendarFilter().filterEvents(
-								masterEventVectorObj.getEvents(
-									getMonthTimeRange((CalendarUtil) AcalObj))));
-					
-					if (!calendarEventVectorObj.isEmpty())
-					{
-						AcalObj.setDay(dateObj1.getYear(),dateObj1.getMonth(),dateObj1.getDay());
-						
-						monthObj2 = calMonth(monthInt, (CalendarUtil)AcalObj, state, calendarEventVectorObj);
-						
-						AcalObj.setDay(dateObj1.getYear(),dateObj1.getMonth(),dateObj1.getDay());
-						
-						if (!calendarEventVectorObj.isEmpty())
-							arrayOfMonths.addElement(monthObj2);
-					}
-				}
-				if (!arrayOfMonths.isEmpty())
-					yearMap.put(Integer.valueOf(yearInt), arrayOfMonths.iterator());
-			}
-		}
-		else
-		{
-			for (yearInt = CalendarFilter.LIST_VIEW_ENDING_YEAR;
-			yearInt >= CalendarFilter.LIST_VIEW_STARTING_YEAR;
-			yearInt--)
-			{
-				Vector arrayOfMonths = new Vector(20);
-				for(monthInt = 12; monthInt >=1; monthInt--)
+		boolean dateAsc = !dateDsc;
+		
+		for (yearInt = dateAsc ? CalendarFilter.LIST_VIEW_STARTING_YEAR : CalendarFilter.LIST_VIEW_ENDING_YEAR;
+			dateAsc ? yearInt <= CalendarFilter.LIST_VIEW_ENDING_YEAR : yearInt >= CalendarFilter.LIST_VIEW_STARTING_YEAR ;
+			yearInt = yearInt + (dateAsc ? 1 : -1)
+		){
+			Vector arrayOfMonths = new Vector(20);
```

Comment:
Avoid using Vector unless it's actually needed

Reply:
I've changed it to an ArrayList,
Vector was there just because I've maintained the inner part of the duplicated for loop.